### DISCUSSION TITLE
캐릭터 움직임 시 사용하는 가상패드 함수나 값은 어떻게 얻을 수 있나요?

### DISCUSSION THREAD

**Discussion by limjemune** [naverz/zepeto-studio-kor#488] [5. Q&A]
안녕하세요. 
캐릭터 움직임 제어 시  키보드, 마우스 액션은 유니티가 제공하는 함수로 쉽게 구현했지만 (Input, InputSystem 등)
제페토 에서 제공하는 케릭터 컨트롤러의 가상패드 의 함수 및 값은 어떻게 구해야하는지 찾질못해서 질문남깁니다.
이유는, 
Move 나 MoveConitnues 로 케릭터가 지속 움직임을 주는 상태에서는 키보드/가상패드의 컨트롤러가 먹통이 되는것같아 따로 컨트롤러 제어 함수를 받아서 제어하려고합니다. 

예) 가상패드.x == 1 (왼쪽)  / 가상패드.x == -1 (오른쪽)

따로 제공하는 함수가 없다면 그냥 일반적인 Touch 로 구하면 되는 것인지? 아니면 InputSystem 의 값을 뽑아서 파싱해야하는건지
아이디어가 필요합니다. 
감사합니다.

**Comment by moondory77-naverz** [comment]
안녕하세요.
우선, 캐릭터가 MoveToPosition / MoveContinuously 상태일 경우
말씀주신것처럼 가상패드를 통한 조작보다 상위에서 이동이 이뤄지기때문에, 가상패드를 통한 조작이 불가합니다.

이 경우 StopMoving() 을 통해 이동을 정지한 다음, 다시 가상패드를 통한 조작이 가능합니다.


**Move 나 MoveConitnues 로 케릭터가 지속 움직임을 주는 상태에서는 키보드/가상패드의 컨트롤러가 먹통이 되는것같아 따로 컨트롤러 제어 함수를 받아서 제어하려고합니다.**

일단 해당 질문에 초점을 맞춰 답변드리자면,
(UIController_Touchpad를 사용할 경우) ZepetoScreenTouchpad 컴포넌트로 접근한다음, 가상패드로의 터치이벤트에 대해 StopMoving()을
호출하는 방법이 있을 듯 합니다.

도움이 될 수 있다면 좋겠습니다.
감사합니다.

**Reply by limjemune** [reply]
답변감사합니다. 
결국 캐릭터가 지속적임 움직임상태에서는 가상패드의 조작은 불가능한게 현재 상태이군요.
캐릭터 가 뛰고있을때  다른 방향을 바꾸는 식으로 단순 제어하려고했지만,  방향을 바꾸려면 
결론적으로,
ZepetoScreenToupad 이벤트가 들어왔을 때 (가상패드) -> StopMoving 으로 제어권을 변경후 ->  캐릭터움직임 변경 -> 다시 Moving 으로 넘기는 식으로 해야하겠군요.
